plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/external/BuildOrderComputer.java

```diff
@@ -0,0 +1,197 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.ui.external;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+
+import org.eclipse.core.resources.IBuildConfiguration;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.n4js.external.N4JSExternalProject;
+import org.eclipse.n4js.internal.N4JSProject;
+import org.eclipse.n4js.projectModel.IN4JSCore;
+import org.eclipse.n4js.projectModel.IN4JSProject;
+import org.eclipse.n4js.projectModel.IN4JSSourceContainerAware;
+import org.eclipse.n4js.ui.external.ComputeProjectOrder.VertexFilter;
+import org.eclipse.n4js.ui.external.ComputeProjectOrder.VertexMapper;
+import org.eclipse.n4js.ui.external.ComputeProjectOrder.VertexOrder;
+import org.eclipse.n4js.ui.internal.N4JSEclipseProject;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Multimap;
+import com.google.inject.Inject;
+
+/**
+ * Utility class for providing build order (as {@link IBuildConfiguration build configuration} instances) for external
+ * projects.
+ */
+public class BuildOrderComputer {
+
+	@Inject
+	private IN4JSCore core;
+
+	private final UriToProjectMapper uri2PrjMapper = new UriToProjectMapper();
+
+	private class UriToProjectMapper implements VertexMapper<String, IN4JSProject> {
+		@Override
+		public IN4JSProject get(String projectURI) {
+			URI uri = org.eclipse.emf.common.util.URI.createURI(projectURI);
+			IN4JSProject project = core.findProject(uri).orNull();
+			return project;
+		}
+
+		@Override
+		public Class<IN4JSProject> getTargetClass() {
+			return IN4JSProject.class;
+		}
+	}
+
+	static private class FilterUnreqestedOut implements VertexFilter<String> {
+		final private Set<String> requestedProjectIds;
+
+		FilterUnreqestedOut(String[] requestedProjectIds) {
+			this.requestedProjectIds = new HashSet<>(Arrays.asList(requestedProjectIds));
+		}
+
+		@Override
+		public boolean matches(String vertex) {
+			boolean filterOut = !requestedProjectIds.contains(vertex);
+			return filterOut;
+		}
+	}
+
+	/**
+	 * Convenience method for {@link #getBuildOrder(N4JSProject[])}
+	 */
+	public VertexOrder<IN4JSProject> getBuildOrder(N4JSExternalProject[] projects) {
+		N4JSProject[] n4jsPrjs = new N4JSProject[projects.length];
+		for (int i = 0; i < projects.length; i++) {
+			N4JSExternalProject project = projects[i];
+			File locationFile = project.getLocation().toFile();
+			URI locationURI = org.eclipse.emf.common.util.URI.createFileURI(locationFile.getAbsolutePath());
+			N4JSProject prjDependency = (N4JSProject) core.findProject(locationURI).orNull();
```

Comment:
I would think that it might be good to add this wrapping logic to the Java-Doc of the find method. In this case here, the returned instance might wrap another project, but it would be ok, I think.